config/facebookPassport.js

```diff
@@ -0,0 +1,50 @@
+const passport = require('passport');
+const User = require('../models/User');
+const FacebookStrategy = require('passport-facebook').Strategy;
+
+// Environment variables
+require('dotenv').config();
+
+// Serialize user saved the providerId to the session
+passport.serializeUser((user, done) => {
+  done(null, user.providerId);
+});
+
+// Deserialize user retrieves the provider id from the session to fine user
+passport.deserializeUser(async (id, done) => {
+  try {
+    const user = await User.findOne({ providerId: id });
+    done(null, user);
+  } catch (err) {
+    console.log('ERROR - facebookAuth.js - deserializeUser', err);
+  }
+});
+
+passport.use(
+  new FacebookStrategy(
+    {
+      clientID: process.env.FACEBOOK_APP_ID,
+      clientSecret: process.env.FACEBOOK_APP_SECRET,
+      callbackURL: '/auth/facebook/redirect'
+    },
+    async (accessToken, refreshToken, profile, done) => {
+      try {
+        const result = await User.findOne({ providerId: profile.id });
+        if (result) {
+          done(null, result);
+        } else {
+          const user = await new User({
+            provider: 'Facebook',
+            providerId: profile.id,
+            username: profile.displayName + profile.id,
+            displayName: profile.displayName,
+            avatar: `https://graph.facebook.com/${profile.id}/picture?type=large`
+          }).save();
+          done(null, user);
+        }
+      } catch (err) {
+        console.log('ERROR - facebookAuth.js - FacebookStrategy', err);
```

Comment:
`next(err)` here as well

Reply:
Next function not working due to Express not being required into file